funnel/forms/proposal.py

```diff
@@ -74,10 +74,18 @@ def __init__(self, *args, **kwargs):
             self.description.description = space.proposal_part_b.get('hint')
 
 
-class ProposalStatusForm(forms.Form):
-    status = forms.SelectField(
-        __("Status"), coerce=int,
-        choices=[(status, label.title) for (status, label) in PROPOSALSTATUS.items() if status != PROPOSALSTATUS.DRAFT])
+class ProposalTransitionForm(forms.Form):
+    transition = forms.SelectField(__("Status"), validators=[forms.validators.DataRequired()])
+
+    def set_queries(self):
+        """
+        value: transition method name
+        label: transition object itself
+        We need the whole object to get the additional metadata in templates
+        """
+        self.transition.choices = sorted([
+            (name, transition) for name, transition in self.edit_obj.state.transitions.items() if hasattr(self.edit_obj.current_access(), name)
```

Comment:
@jace I dont know if it's an issue or design, but `edit_obj.state.transitions.items()` is showing transitions the current user role doesn't have access to. E.g. `awaiting_details` is available to only `reviewer`, but `admin` can still see it in items(). But of course `current_access()` doesn't have it.

Reply:
This is a design issue. RoleMixin and StateManager aren't aware of each other, so they don't offer any way to obtain just the intersection. My guess is StateManager should offer an additional `transitions_for` method.

Reply:
Ok. So can we go with this for now? And update it once the changes in statemanager is made?